jlm/tooling/Makefile.sub

```diff
@@ -57,10 +57,13 @@ libtooling_TEST_LIBS = \
 	libtooling \
 	libhls \
 	libllvm \
+	libmlir \
 	librvsdg \
 	libutil \
 	libjlmtest \
 
-libtooling_TEST_EXTRA_LDFLAGS = $(shell $(LLVMCONFIG) --ldflags --libs --system-libs)
+libtooling_TEST_EXTRA_LDFLAGS = \
+	$(shell $(LLVMCONFIG) --ldflags --libs --system-libs) \
+	${MLIR_LDFLAGS} \
```

Comment:
The flags have always been there for the LLVM dialect as part of the LLVMCONFIG.

That they are not needed for HLS is likely because the HLS backend has dedicated tools, i.e., jlm-hls and jhls, so when linking jlm-opt and jlc you never have a need to link against the HLS libraries.

Reply:
You misunderstood my question. Why do you need all in a sudden hand in the LLVM linker flags for the libtooling? The libtooling was already dependent on LLVM before the MLIR dialect came along, and it compiled just fine without these flags.

Does it really not compile any longer when you remove these flags?